### PR TITLE
Initial code for the OCI Registry functionality: auth and hit the server.

### PR DESCRIPTION
**Pull request by facundobatista**

### CODE REVIEW

File: charmcraft/commands/store/registry.py [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+from urllib.request import parse_http_list, parse_keqv_list
+
+import requests
+
+from charmcraft.cmdbase import CommandError
+
+logger = logging.getLogger(__name__)
+
+# some mimetypes
+MANIFEST_LISTS = 'application/vnd.docker.distribution.manifest.list.v2+json'
```
Comment (cjp256) [inline charmcraft/commands/store/registry.py:29]: Googled this and found surprisingly few hits. Not a friendly search term I guess :rofl: Found some docs @ https://docs.docker.com/registry/spec/manifest-v2-2/

**Review by cjp256 — APPROVED** [review]
LGTM

File: charmcraft/commands/store/registry.py (hunk 1 above)
Comment (facundobatista) [reply charmcraft/commands/store/registry.py:29]: yes, those are the ONLY docs...

**Review by cmatsuoka — APPROVED** [review]
Looks good.

**Pull request merged**